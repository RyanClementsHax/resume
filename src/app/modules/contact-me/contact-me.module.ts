import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { Routes, RouterModule } from '@angular/router';

import { ContactMeComponent } from './containiers/contact-me/contact-me.component';
import { SocialMediaLinkComponent } from './components/social-media-link/social-media-link.component';

export const ROUTES: Routes = [
  { path: '', component: ContactMeComponent }
];

@NgModule({
  imports: [
    CommonModule,
    RouterModule.forChild(ROUTES)
  ],
  declarations: [ContactMeComponent, SocialMediaLinkComponent]
})
export class ContactMeModule {}
