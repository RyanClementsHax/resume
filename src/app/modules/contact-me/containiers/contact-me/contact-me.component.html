<header class="jumbotron">
  <h1>Just yell <span class="quote">"RYAN!!"</span> really loudly</h1>
  <p>or...</p>
</header>
<main>
  <article>
    <h2>Let's get social</h2>
    <social-media-link></social-media-link>
  </article>
</main>