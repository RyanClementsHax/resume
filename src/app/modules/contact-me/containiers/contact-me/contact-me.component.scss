@import "variables.scss";

.jumbotron {
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    text-align: center;
    height: $jumbotron-height;
    padding: $content-padding;
    background: $color-primary-faded;

    h1 {
        margin: 0;

        .quote {
            color: $color-accent;
        }
    }

    p {
        position: absolute;
        margin: 0 auto;
        bottom: 15px;
    }
}

main {
    @media screen and (min-width: $small-screen-break-point) {
        height: $min-page-height;
        display: flex;
        flex-wrap: wrap;
        justify-content: space-around;
        align-items: center;
        align-content: center;
    }

    .social-link {
        display: flex;
        justify-content: center;
        align-items: center;
        padding: 70px 20px;
    
        h2 {
            display: inline-block;
            margin: 0;
            margin-left: 30px;
            font-size: 3em;
        }
    
        h2, i {
            color: white;
        }
    
        &.github-link {
            background: #24292e;
        }
    
        &.twitter-link {
            background: #1da1f2;
        }
    
        &.linkedin-link {
            background: #0077b5;
        }
    
        &.gmail-link {
            background: #d93025;
        }
    
        &:hover {
            box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.3);
        }

        @media screen and (min-width: $small-screen-break-point) {
            padding: 20px 50px;
            margin-top: 20px;
            border-radius: 50px;
        }
    }
}