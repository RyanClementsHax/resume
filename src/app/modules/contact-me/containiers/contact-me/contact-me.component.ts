import { Component } from '@angular/core';

@Component({
  selector: 'app-contact-me',
  templateUrl: './contact-me.component.html',
  styleUrls: ['./contact-me.component.scss']
})
export class ContactMeComponent {}
