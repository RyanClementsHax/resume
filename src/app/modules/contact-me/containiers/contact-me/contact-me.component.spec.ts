import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';

import { ContactMeComponent } from './contact-me.component';

describe('ContactMeComponent', () => {
  let component: ContactMeComponent;
  let fixture: ComponentFixture<ContactMeComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ContactMeComponent],
      schemas: [NO_ERRORS_SCHEMA]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ContactMeComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
