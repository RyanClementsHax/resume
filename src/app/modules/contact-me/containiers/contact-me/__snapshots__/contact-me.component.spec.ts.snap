// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactMeComponent should match snapshot 1`] = `ContactMeComponent {}`;
